src/app/messagefilter.cpp

```diff
@@ -92,7 +94,15 @@ bool MessageFilter::filterAcceptsRow(int sourceRow, const QModelIndex& sourcePar
         case IrcMessage::Topic:
         case IrcMessage::Numeric:
         case IrcMessage::Names:
-            return m_topicMessages || !dynamic_cast<const MessageModel*>(sourceModel())->buffer()->isChannel();
+            if (m_topicMessages)
+            {
+                if (m_buffer->connection()->connectionCount() == 1)
+                    return false;
+                else
+                    return true;
+            }
+            else
+                return m_buffer->isChannel();
```

Comment:
the logic is a bit confusing here, but wasn't it `!isChannel()` before?

Reply:
previously, topics/numerics/names were shown if either:
- topic messages were enabled in the ui (m_topicMessages), or
- the buffer is not a channel buffer (but e.g. the server buffer, private query with a bouncer, or such)

presumably we want to keep the logic otherwise the same, but exclude topic/numeric/names messages when reconnecting, as in, when the connection count is > 1?

by that logic the condition would be something like:

```
case IrcMessage::Topic:
case IrcMessage::Numeric:
case IrcMessage::Names:
    return !m_buffer->isChannel() || (m_topicMessages && m_buffer->connection()->connectionCount() <= 1);
```